packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -198,14 +198,25 @@ class DependencyExtractionWebpackPlugin {
 				}
 			}
 
+			// Go through the assets and hash the sources. We can't just use
+			// `entrypointChunk.contentHash` because that's not updated when
+			// assets are minified. Sigh.
+			// @todo Use `asset.info.contenthash` if we can make sure it's reliably set.
+			const hash = createHash( 'sha512' );
+			for ( const filename of entrypoint.getFiles().sort() ) {
+				const asset = compilation.getAsset( filename );
```

Comment:
It looks like we could simplify the code here to:

```js
hash.update( filename + ':' + compilation.getAsset( filename ).source.source() );
```

Reply:
Only in that `asset.source.updateHash()` may be more efficient, processing the internal representation of the source directly instead of having to stringify it for concatenation first.

Reply:
Right, that could be the case.